### PR TITLE
:all_pages option, collapse api method signatures, configure resultsperpage, builtin filters

### PR DESCRIPTION
**Pull request by apurvis**
@jarthod 
OK so this supersedes PRs 42 and 43 and has been tested in production.

**Changes**

1. deprecate the 2nd hash param in `.all` and `.first`
2. enable configuration of a default value for the `resultsperpage` parameter. Setting this higher might help with some people's timeout issues, depending on how they are making requests. I left the gem default as the SG default.
3. an optional condition `:all_pages` for the `.all` method so that `SurveyGizmo::API::Survey.all(all_pages: true)` really gives you all your surveys (we just hit a bug where we went over 50 active surveys and didn't realize `SurveyGizmo::API::Survey.all` wasn't returning them all)
4. a bunch of built in filters that i moved over from our application to make them reusable

you've seen this all before except number 4.

i would suggest releasing this as 4.1.0?

### CODE REVIEW

**Comment by apurvis** [comment]
merge both my commit and https://github.com/RipTheJacker/survey-gizmo-ruby/pull/41 and release them both as 4.1.0?

File: README.md [hunk 1]
```diff
@@ -68,14 +73,29 @@ question.save
...
+# Retrieving page 3 of completed, non test data SurveyResponses submitted within the past 3 days
+# for contact id 999.  (This example shows you how to use some of the gem's built in filters and
+# filtergenerators and also how to construct your own raw filter hash)
+#
+# See: http://apihelp.surveygizmo.com/help/article/link/filters for more info on filters
+responses = SurveyGizmo::API::Response.all(
+  survey_id: 12345,
+  page: 3,
+  filters: [
+    SurveyGizmo::API::Response::NO_TEST_DATA,
```
Comment (apurvis) [inline README.md]: here's the docs/examples on the built in filters

**Comment by apurvis** [comment]
threw some more cosmetic changes on there; it's all working in production

**Comment by jarthod** [comment]
Looks nice, thanks!

**Comment by jarthod** [comment]
Released `4.1.0` :beers:

**Pull request merged**